python/grass/tools/session_tools.py

```diff
@@ -17,10 +17,82 @@
 import os
 
 import grass.script as gs
+from grass.exceptions import CalledModuleError
 
 from .support import ParameterConverter, ToolFunctionResolver, ToolResult
 
 
+class ToolError(CalledModuleError):
+    """Raised when a tool run ends with error (typically a non-zero return code)
+
+    Inherits from *subprocess.CalledProcessError* to make it easy to transition from
+    or combine with code which is using the *subprocess* package.
+    Inherits from CalledModuleError to make it easy to transition code with except
+    statements around *grass.script.run_command*-style tool calls, but new code should
+    not rely on that.
+    """
+
+    def __init__(self, tool, cmd, returncode, errors=None):
+        """Create an exception with a full error message based on the parameters.
+
+        Best results are provided when errors is a single line string, aiming at a
+        short and clear message. In case it is `None`, additional text is providing
```

Comment:
The single backtick in Sphinx (ReST) does not mean the same that in Markdown

Reply:
...but it can mean the same thing as a fall back if configured that way. I see our default_role as `# default_role = None`. But changing to `default_role = "literal"` or `default_role = "code"` seems to be the solution here. While in another world, having the default role as `py:obj` or `any` seems good, with all the Markdown around us, having backticks as literal/code is the only option. What do you think?

Reply:
I went ahead and implemented that in #6399.